client/src/Pages/AddReviewPage/AddReviewPage.tsx

```diff
@@ -0,0 +1,113 @@
+import { useEffect, useState } from 'react';
+import { useHistory, useParams } from 'react-router';
+import { useSelector } from 'react-redux';
+import { useToasts } from 'react-toast-notifications';
+import { RootState } from '../../app/store';
+import { Axios } from '../../helpers/axios';
+import { format } from 'date-fns';
+import { BigLabel } from '../../Components/BigLabel/BigLabel';
+import { BookDescription } from '../../Components/BookDescription/BookDescription';
+import StarRating from '../../Components/Star/StarRating';
+import { Button } from '../../Components/Button/Button';
+import {
+  Flex,
+  Title,
+  Description,
+  Rating,
+  Text,
+  Review,
+  ReviewInput,
+  AddReview,
+} from '../AddReviewPage/AddReviewStyles';
+
+type Book = {
+  id: number;
+  isbn: number;
+  title: string;
+  description?: string;
+  release_date: string;
+  num_pages?: number;
+  cover?: string;
+  author_id?: number;
+  name?: string;
+  surname?: string;
+  average?: number;
+};
+
+export const AddReviewPage = () => {
+  const { id } = useParams<{ id: string }>();
+  const [bookData, setBookData] = useState<Book>();
+  const [userReview, setUserReview] = useState<string>('');
+  const [userStars, setUserStars] = useState<number>(0);
+  const { addToast } = useToasts();
+  const history = useHistory();
+  // Czy to tak sie uzywa useSelector by pobrac id ze store?
+  const userId = useSelector((state: RootState) => state.user.user?.id);
+
+  const handleClick = () => {
+    const date = format(new Date(), 'YYYY-MM-DD HH:mm:ss');
+    const reviewInfo = {
+      content: userReview,
+      header: 'test',
+      score: userStars,
+      added_at: date,
+      user_id: userId,
+      book_id: id,
+    };
+
+    Axios.post('http://localhost:8080/api/reviews/', reviewInfo).then(() => {
+      history.push('/');
+      addToast(`Review added for ${bookData?.title}`, {
+        appearance: 'success',
+        autoDismiss: true,
+      });
+    });
+  };
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const bookResult: any = await Axios('http://localhost:8080/api/books/details/' + id);
+      const rateResult: any = await Axios('http://localhost:8080/api/books/average/' + id);
+      if (rateResult.data.length > 0) {
+        const average = parseInt(rateResult.data[0].rating);
+        setBookData({ ...bookResult.data[0], average });
+      } else {
+        setBookData(bookResult.data[0]);
+      }
+    };
+    fetchData();
+  }, [id]);
+
+  return (
+    <Flex>
+      <Title>
+        <BigLabel title={bookData?.title ?? 'Loading...'}></BigLabel>
```

Comment:
TO loading występuje w trakcie fetchowania danych?
Może warto jakiś spinner dodać albo chodziaż info może //TO DO na kiedyś :D